[BUG] Codebase test failures in Windows

After #11618 I'm seeing failure of all the import tests. I verified the issue with @tcmetzger



```
pytest tests/codebase/test_no_ipython_common.py
```

#### Stack traceback and/or browser JavaScript console output
```
(bkdev) C:\Users\parki\Documents\GitHub\bokeh>pytest tests/codebase/test_no_ipython_common.py
================================================================================================================= test session starts ==================================================================================================================
platform win32 -- Python 3.9.7, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
rootdir: C:\Users\parki\Documents\GitHub\bokeh, configfile: setup.cfg
plugins: flaky-3.7.0, asyncio-0.15.1, cov-2.12.1, forked-1.3.0, html-3.1.1, metadata-1.11.0, xdist-2.3.0
collected 1 item

tests\codebase\test_no_ipython_common.py F                                                                                                                                                                                                        [100%]

======================================================================================================================= FAILURES ======================================================================================================================= 
___________________________________________________________________________________________________________ test_no_ipython_common_combined ____________________________________________________________________________________________________________ 

    def test_no_ipython_common_combined() -> None:
        ''' Basic usage of Bokeh should not result in any IPython code being
        imported. This test ensures that importing basic modules does not bring in
        IPython.

        '''
        proc = run([python, "-c", verify_clean_imports('IPython', MODULES)])
>       assert proc.returncode == 0, "IPython imported in common modules"
E       AssertionError: IPython imported in common modules
E       assert 1 == 0
E        +  where 1 = CompletedProcess(args=['C:\\Users\\parki\\anaconda3\\envs\\bkdev\\python.exe', '-c', "import sys; import bokeh/__init_...sion;import bokeh/util/warnings; sys.exit(1 if any('IPython' in x for x in sys.modules.keys()) else 0)"], 
returncode=1).returncode

tests\codebase\test_no_ipython_common.py:52: AssertionError
----------------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------------- 
  File "<string>", line 1
    import sys; import bokeh/__init__;import bokeh/_testing/__init__;import bokeh/_testing/plugins/__init__;import bokeh/_testing/plugins/bokeh_server;import bokeh/_testing/plugins/file_server;import bokeh/_testing/plugins/ipython;import bokeh/_testing/plugins/jupyter_notebook;import bokeh/_testing/plugins/log_file;import bokeh/_testing/plugins/managed_server_loop;import bokeh/_testing/plugins/networkx;import bokeh/_testing/plugins/pandas;import bokeh/_testing/plugins/project;import bokeh/_testing/plugins/selenium;import bokeh/_testing/util/__init__;import bokeh/_testing/util/api;import bokeh/_testing/util/compare;import bokeh/_testing/util/env;import bokeh/_testing/util/examples;import bokeh/_testing/util/filesystem;import bokeh/_testing/util/project;import bokeh/_testing/util/screenshot;import bokeh/_testing/util/selenium;import bokeh/_testing/util/types;import bokeh/_version;import bokeh/application/__init__;import bokeh/application/application;import bokeh/application/handlers/__init__;import bokeh/application/handlers/code;import bokeh/application/handlers/code_runner;import bokeh/application/handlers/directory;import bokeh/application/handlers/document_lifecycle;import bokeh/application/handlers/function;import bokeh/application/handlers/handler;import bokeh/application/handlers/lifecycle;import bokeh/application/handlers/notebook;import bokeh/application/handlers/request_handler;import bokeh/application/handlers/script;import bokeh/application/handlers/server_lifecycle;import bokeh/application/handlers/server_request_handler;import bokeh/client/__init__;import bokeh/client/connection;import bokeh/client/session;import bokeh/client/states;import bokeh/client/util;import bokeh/client/websocket;import bokeh/colors/__init__;import bokeh/colors/color;import bokeh/colors/groups;import bokeh/colors/hsl;import bokeh/colors/named;import bokeh/colors/rgb;import bokeh/colors/util;import bokeh/command/__init__;import bokeh/command/bootstrap;import bokeh/command/subcommand;import bokeh/command/subcommands/__init__;import bokeh/command/subcommands/build;import bokeh/command/subcommands/file_output;import bokeh/command/subcommands/info;import bokeh/command/subcommands/init;import bokeh/command/subcommands/json;import bokeh/command/subcommands/sampledata;import bokeh/command/subcommands/secret;import bokeh/command/subcommands/serve;import bokeh/command/subcommands/static;import bokeh/command/util;import bokeh/core/__init__;import bokeh/core/enums;import bokeh/core/has_props;import bokeh/core/json_encoder;import bokeh/core/properties;import bokeh/core/property/__init__;import bokeh/core/property/_sphinx;import bokeh/core/property/alias;import bokeh/core/property/any;import bokeh/core/property/auto;import bokeh/core/property/bases;import bokeh/core/property/color;import bokeh/core/property/container;import bokeh/core/property/dataspec;import bokeh/core/property/datetime;import bokeh/core/property/descriptor_factory;import bokeh/core/property/descriptors;import bokeh/core/property/either;import bokeh/core/property/enum;import bokeh/core/property/factors;import bokeh/core/property/include;import bokeh/core/property/instance;import bokeh/core/property/json;import bokeh/core/property/nullable;import bokeh/core/property/numeric;import bokeh/core/property/override;import bokeh/core/property/pandas;import bokeh/core/property/primitive;import bokeh/core/property/readonly;import bokeh/core/property/singletons;import bokeh/core/property/string;import bokeh/core/property/struct;import bokeh/core/property/text_like;import bokeh/core/property/validation;import bokeh/core/property/visual;import bokeh/core/property/wrappers;import bokeh/core/property_mixins;import bokeh/core/query;import bokeh/core/templates;import bokeh/core/types;import bokeh/core/validation/__init__;import bokeh/core/validation/check;import bokeh/core/validation/decorators;import bokeh/core/validation/errors;import bokeh/core/validation/issue;import bokeh/core/validation/warnings;import bokeh/document/__init__;import bokeh/document/callbacks;import bokeh/document/document;import bokeh/document/events;import bokeh/document/json;import bokeh/document/locking;import bokeh/document/models;import bokeh/document/modules;import bokeh/document/util;import bokeh/driving;import bokeh/embed/__init__;import bokeh/embed/bundle;import bokeh/embed/elements;import bokeh/embed/notebook;import bokeh/embed/server;import bokeh/embed/standalone;import bokeh/embed/util;import 
bokeh/embed/wrappers;import bokeh/events;import bokeh/ext;import bokeh/io/__init__;import bokeh/io/doc;import bokeh/io/export;import bokeh/io/notebook;import bokeh/io/output;import bokeh/io/saving;import bokeh/io/showing;import bokeh/io/state;import bokeh/io/util;import bokeh/io/webdriver;import bokeh/layouts;import bokeh/model/__init__;import bokeh/model/data_model;import bokeh/model/docs;import bokeh/model/model;import bokeh/model/util;import bokeh/models/__init__;import bokeh/models/annotations;import bokeh/models/arrow_heads;import bokeh/models/axes;import bokeh/models/callbacks;import bokeh/models/canvas;import bokeh/models/css;import bokeh/models/dom;import bokeh/models/expressions;import bokeh/models/filters;import bokeh/models/formatters;import bokeh/models/glyph;import bokeh/models/glyphs;import bokeh/models/graphs;import bokeh/models/grids;import bokeh/models/labeling;import bokeh/models/layouts;import bokeh/models/map_plots;import bokeh/models/mappers;import bokeh/models/plots;import bokeh/models/ranges;import bokeh/models/renderers;import bokeh/models/scales;import bokeh/models/selections;import bokeh/models/sources;import bokeh/models/text;import bokeh/models/textures;import bokeh/models/tickers;import bokeh/models/tiles;import bokeh/models/tools;import bokeh/models/transforms;import bokeh/models/util/__init__;import bokeh/models/util/structure;import bokeh/models/widgets/__init__;import bokeh/models/widgets/buttons;import bokeh/models/widgets/groups;import bokeh/models/widgets/icons;import bokeh/models/widgets/inputs;import bokeh/models/widgets/markups;import bokeh/models/widgets/panels;import bokeh/models/widgets/sliders;import bokeh/models/widgets/tables;import bokeh/models/widgets/widget;import bokeh/palettes;import bokeh/plotting/__init__;import bokeh/plotting/_decorators;import bokeh/plotting/_docstring;import bokeh/plotting/_graph;import bokeh/plotting/_legends;import bokeh/plotting/_plot;import bokeh/plotting/_renderer;import bokeh/plotting/_stack;import bokeh/plotting/_tools;import bokeh/plotting/figure;import bokeh/plotting/glyph_api;import bokeh/plotting/gmap;import bokeh/plotting/graph;import bokeh/protocol/__init__;import bokeh/protocol/exceptions;import bokeh/protocol/message;import bokeh/protocol/messages/__init__;import bokeh/protocol/messages/ack;import bokeh/protocol/messages/error;import bokeh/protocol/messages/ok;import bokeh/protocol/messages/patch_doc;import bokeh/protocol/messages/pull_doc_reply;import bokeh/protocol/messages/pull_doc_req;import bokeh/protocol/messages/push_doc;import bokeh/protocol/messages/server_info_reply;import bokeh/protocol/messages/server_info_req;import bokeh/protocol/receiver;import bokeh/resources;import bokeh/sampledata/__init__;import bokeh/sampledata/airport_routes;import bokeh/sampledata/airports;import bokeh/sampledata/anscombe;import bokeh/sampledata/antibiotics;import bokeh/sampledata/autompg;import bokeh/sampledata/autompg2;import bokeh/sampledata/browsers;import 
bokeh/sampledata/commits;import bokeh/sampledata/daylight;import bokeh/sampledata/degrees;import bokeh/sampledata/gapminder;import bokeh/sampledata/glucose;import bokeh/sampledata/haar_cascade;import bokeh/sampledata/iris;import bokeh/sampledata/les_mis;import bokeh/sampledata/movies_data;import bokeh/sampledata/mtb;import bokeh/sampledata/olympics2014;import bokeh/sampledata/penguins;import bokeh/sampledata/perceptions;import bokeh/sampledata/periodic_table;import bokeh/sampledata/population;import bokeh/sampledata/sample_geojson;import bokeh/sampledata/sea_surface_temperature;import bokeh/sampledata/sprint;import bokeh/sampledata/stocks;import bokeh/sampledata/unemployment;import bokeh/sampledata/unemployment1948;import bokeh/sampledata/us_cities;import bokeh/sampledata/us_counties;import bokeh/sampledata/us_holidays;import bokeh/sampledata/us_marriages_divorces;import bokeh/sampledata/us_states;import bokeh/sampledata/world_cities;import bokeh/server/__init__;import bokeh/server/auth_provider;import bokeh/server/callbacks;import bokeh/server/connection;import bokeh/server/contexts;import bokeh/server/django/__init__;import bokeh/server/django/apps;import bokeh/server/django/consumers;import bokeh/server/django/routing;import bokeh/server/django/static;import bokeh/server/protocol_handler;import bokeh/server/server;import bokeh/server/session;import bokeh/server/tornado;import bokeh/server/urls;import bokeh/server/util;import bokeh/server/views/__init__;import bokeh/server/views/auth_request_handler;import bokeh/server/views/autoload_js_handler;import bokeh/server/views/doc_handler;import bokeh/server/views/metadata_handler;import bokeh/server/views/multi_root_static_handler;import bokeh/server/views/root_handler;import bokeh/server/views/session_handler;import bokeh/server/views/static_handler;import bokeh/server/views/ws;import bokeh/settings;import bokeh/sphinxext/__init__;import bokeh/sphinxext/bokeh_autodoc;import bokeh/sphinxext/bokeh_color;import bokeh/sphinxext/bokeh_dataframe;import bokeh/sphinxext/bokeh_directive;import bokeh/sphinxext/bokeh_enum;import bokeh/sphinxext/bokeh_example_metadata;import bokeh/sphinxext/bokeh_gallery;import bokeh/sphinxext/bokeh_jinja;import bokeh/sphinxext/bokeh_model;import bokeh/sphinxext/bokeh_options;import bokeh/sphinxext/bokeh_palette;import bokeh/sphinxext/bokeh_palette_group;import bokeh/sphinxext/bokeh_plot;import bokeh/sphinxext/bokeh_prop;import bokeh/sphinxext/bokeh_releases;import bokeh/sphinxext/bokeh_roles;import bokeh/sphinxext/bokeh_settings;import bokeh/sphinxext/bokeh_sitemap;import bokeh/sphinxext/bokehjs_content;import bokeh/sphinxext/collapsible_code_block;import bokeh/sphinxext/example_handler;import bokeh/sphinxext/sample;import bokeh/sphinxext/templates;import bokeh/sphinxext/util;import bokeh/themes/__init__;import bokeh/themes/_caliber;import bokeh/themes/_contrast;import bokeh/themes/_dark_minimal;import bokeh/themes/_light_minimal;import bokeh/themes/_night_sky;import bokeh/themes/theme;import bokeh/tile_providers;import bokeh/transform;import bokeh/util/__init__;import bokeh/util/browser;import bokeh/util/callback_manager;import bokeh/util/compiler;import bokeh/util/dataclasses;import bokeh/util/datatypes;import bokeh/util/dependencies;import bokeh/util/deprecation;import bokeh/util/functions;import bokeh/util/hex;import bokeh/util/logconfig;import bokeh/util/options;import bokeh/util/paths;import bokeh/util/sampledata;import bokeh/util/serialization;import bokeh/util/string;import bokeh/util/terminal;import bokeh/util/token;import bokeh/util/tornado;import bokeh/util/version;import bokeh/util/warnings; sys.exit(1 if any('IPython' in x for x in sys.modules.keys()) else 0)
                            ^
SyntaxError: invalid syntax
=================================================================================================================== warnings summary ===================================================================================================================
..\..\..\anaconda3\envs\bkdev\lib\site-packages\pyreadline\py3k_compat.py:8
  C:\Users\parki\anaconda3\envs\bkdev\lib\site-packages\pyreadline\py3k_compat.py:8: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working       
    return isinstance(x, collections.Callable)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=============================================================================================================== short test summary info ================================================================================================================
FAILED tests/codebase/test_no_ipython_common.py::test_no_ipython_common_combined - AssertionError: IPython imported in common modules
============================================================================================================= 1 failed, 1 warning in 0.34s =============================================================================================================
```
#### Screenshots or screencasts of the bug in action

*Note: Everyone interacting in the Bokeh project's codebases, issue trackers, and discussion forums is expected to follow the [Code of Conduct](https://github.com/bokeh/bokeh/blob/master/CODE_OF_CONDUCT.md).*

--- a/bokeh/_testing/util/project.py
+++ b/bokeh/_testing/util/project.py
@@ -21,7 +21,6 @@ log = logging.getLogger(__name__)
 #-----------------------------------------------------------------------------
 
 # Standard library imports
-import os
 from pathlib import Path
 from subprocess import run
 from typing import List, Sequence
@@ -59,7 +58,7 @@ def ls_modules(*, dir: str = "bokeh", skip_prefixes: Sequence[str] = [], skip_ma
         if file.endswith("__main__.py") and skip_main:
             continue
 
-        module = file.replace(os.sep, ".").replace(".py", "").replace(".__init__", "")
+        module = file.replace("/", ".").replace(".py", "").replace(".__init__", "")
 
         if any(module.startswith(prefix) for prefix in skip_prefixes):
             continue

